ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,316 @@
+        progressIndicator.setVisible(false);
+        applyQueryButton.setDisable(false);
+        // display number of features returned
+        setFeatureNumberLabelText();
+      });
+
+      // handle UI
+      revertToInitialQueryButton.setDisable(true);
+      textField.setText("3000");
+      // reset combo box and show original prompt text
+      comboBox.getSelectionModel().clearSelection();
+      comboBox.setButtonCell(new ListCell<>() {
+        @Override
+        protected void updateItem(String item, boolean empty) {
+          setText(item);
+        }
+      });
+
+    } catch (Exception exception) {
+      exception.printStackTrace();
+      new Alert(Alert.AlertType.ERROR, exception.getMessage()).show();
+    }
+  }
+
+  /**
+   * Populates features from provided query parameters, and displays the result on the map.
+   */
+  @FXML
+  private void query() {
+
+    QueryParameters cqlQueryParameters = new QueryParameters();
+    // set the query parameter's where clause with the CQL query in the combo box
+    cqlQueryParameters.setWhereClause(comboBox.getSelectionModel().getSelectedItem());
+    // set the max features to the number entered in the text field
+    cqlQueryParameters.setMaxFeatures(Integer.parseInt(textField.getText()));
+
+    // if the time extent checkbox is selected, retrieve the date selected from the date picker and set it to the
+    // query parameters time extent
+    if (timeExtentCheckBox.isSelected()) {
+      // get the value selected from the date picker
+      LocalDate startDate = startDatePicker.getValue();
+      LocalDate endDate = endDatePicker.getValue();
+
+      // convert the value into a calendar
+      Calendar start = new Calendar.Builder().setDate(startDate.getYear(), startDate.getMonthValue(), startDate.getDayOfMonth()).build();
+      Calendar end = new Calendar.Builder().setDate(endDate.getYear(), endDate.getMonthValue(), endDate.getDayOfMonth()).build();
+
+      // set the query parameters time extent
+      cqlQueryParameters.setTimeExtent(new TimeExtent(start, end));
+    }
+
+    // populate and load the table with the query parameters
+    // set the clearCache parameter to true to clear existing table entries and set the outfields parameter to null requests all fields
+    ListenableFuture<FeatureQueryResult> result = ogcFeatureCollectionTable.populateFromServiceAsync(cqlQueryParameters, true, null);
+    result.addDoneListener(() -> {
+
+      // display number of features returned
+      setFeatureNumberLabelText();
+
+      try {
+        // create a new list to store returned geometries in
+        List<Geometry> featureGeometryList = new ArrayList<>();
+
+        // iterate through each result to get its geometry and add it to the geometry list
+        result.get().iterator().forEachRemaining(feature -> {
+          featureGeometryList.add(feature.getGeometry());
+          feature.getGeometry();
```

Comment:
Don't you get a warning here about not using the return value? Actually do you need this line since you already got and used the geometry on the line above?

Reply:
good spot: IDE gave no warning! I've removed that line.